clippy_lints/src/methods/mod.rs

```diff
@@ -3314,6 +3340,25 @@ fn lint_single_char_push_string(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args
     }
 }
 
+/// lint for length-1 `str`s as argument for `insert_str`
+fn lint_single_char_insert_string(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args: &[hir::Expr<'_>]) {
+    let mut applicability = Applicability::MachineApplicable;
+    if let Some(extension_string) = get_hint_if_single_char_arg(cx, &args[2], &mut applicability) {
+        let base_string_snippet = snippet_with_applicability(cx, args[0].span, "_", &mut applicability);
```

Comment:
I was just passing by fixed issue... It works poorly with macro (e.g. `dbg!(s).push_str()`). Do we need to use one of the `snippet_with_macro` variant?

Reply:
Oh this is a good point, thank you!